ui/shared/JobInfo.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { getInspectTaskUrl, getJobSearchStrHref } from '../helpers/url';
+import { getSearchStr, getTimeFields } from '../helpers/job';
+
+export default class JobInfo extends React.PureComponent {
+  render() {
+    const { job, jobLogUrls, jobDetails } = this.props;
+
+    const timeFields = getTimeFields(job);
+    const jobSearchStr = getSearchStr(job);
+    const iconCircleClass = job.job_type_description ? 'fa fa-info-circle' : '';
+
+    return (
+      <div>
+        <li className="small">
+          <label title="">Job</label>
+          <a
+            title="Filter jobs with this unique SHA signature"
+            href={getJobSearchStrHref(job.signature)}
+          >(sig)</a>:&nbsp;
+          <a
+            title="Filter jobs containing these keywords"
+            href={getJobSearchStrHref(jobSearchStr)}
+          >{jobSearchStr}</a>
+        </li>
+        {job.taskcluster_metadata &&
+          <li className="small">
+            <label>Task: </label>
+            <a
+              href={getInspectTaskUrl(job.taskcluster_metadata.task_id)}
+              target="_blank"
+              rel="noopener noreferrer"
+            >{job.taskcluster_metadata.task_id}</a>
+          </li>
+        }
+        <li className="small">
+          <label>Build: </label>
+          <span>{`${job.build_architecture} ${job.build_platform} ${job.build_os || ''}`}</span>
+          <span className={`ml-1 ${iconCircleClass}`} />
+        </li>
+        <li className="small">
+          <label>Job name: </label>
+          <span>{job.job_type_name}</span>
+          <span className={`ml-1${iconCircleClass}`} />
+        </li>
+        {timeFields && <span>
+          <li className="small">
+            <label>Requested: </label>{timeFields.requestTime}
+          </li>
+          {timeFields.startTime && <li className="small">
+            <label>Started: </label>{timeFields.startTime}
+          </li>}
+          {timeFields.endTime && <li className="small">
+            <label>Ended: </label>{timeFields.endTime}
+          </li>}
+          <li className="small">
+            <label>Duration: </label>{timeFields.duration}
+          </li>
+        </span>}
+        {!jobLogUrls.length ?
+          <li className="small"><label>Log parsing status: </label>No logs</li> :
+          jobLogUrls.map(data => (
+            <li className="small" key={data}>
+              <label>Log parsing status: </label>{data.parse_status}
+            </li>
+          ))
+        }
+        {jobDetails.map(detail => (<li>
```

Comment:
This array isn't used in Treeherder, but will be used in the new LogViewer.

Reply:
Could the log viewer re-use parts of `JobDetailsTab` instead? If not I wonder if it's best to keep this separate, given it's only used in the log viewer and not elsewhere when JobInfo is used?

Reply:
This is a good point.  I moved the ``JobDetailsTab`` to be named just ``JobDetails`` and can use that in the ``Logviewer`` later.  But also allowed this component to take an ``extraFields`` array.  I unified that shape of the "time" fields and this ``extraFields`` so one loop can handle al of them.  I think this is a nice way to allow it to expand, and also reduce code.  I'm curious if you'll agree.  :)

I moved the ``getTimeFields`` function into the ``JobInfo`` file, since that's the only place that was used anyway.

As we discussed in IRC, I removed a couple crufty buildbot pieces, too.